### ISSUE TITLE
Can you explain how the workflow is found? 

### ISSUE THREAD

**Issue by chenghuang-mdsol** [felixp8/dispatch-and-wait#1]
I see some operations around workflow created_at and request created time. Is it just by time check?

**Comment by felixp8** [comment]
Yes, it is kind of messy, but `find_workflow` is trying to check if the latest workflow returned by the cURL request was triggered within the last 10 seconds. It often takes a few seconds for a recently triggered workflow to appear on the repository dispatch page, so that step was to guarantee that the workflow that is monitored is the one that was just triggered. This assumes that workflows are triggered at least 10 seconds apart, which is fine for the project I was working on, but might not be true in general.

zigiai's [fork](https://github.com/zigiai/dispatch-and-wait) has a cleaner and more reliable method that gets the expected workflow id before triggering the new workflow. I can try to add something similar, but I've been quite busy lately so I may not get to it any time soon.

**Comment by chenghuang-mdsol** [comment]
Copied that. I read the github actions workflow APIs. I didn't find a good way to sync with triggered workflows.
I guess for now we just check the workflow within the last 10 secs.
Thank you for the confirmation.

> Yes, it is kind of messy, but `find_workflow` is trying to check if the latest workflow returned by the cURL request was triggered within the last 10 seconds. It often takes a few seconds for a recently triggered workflow to appear on the repository dispatch page, so that step was to guarantee that the workflow that is monitored is the one that was just triggered. This assumes that workflows are triggered at least 10 seconds apart, which is fine for the project I was working on, but might not be true in general.
> 
> zigiai's [fork](https://github.com/zigiai/dispatch-and-wait) has a cleaner and more reliable method that gets the expected workflow id before triggering the new workflow. I can try to add something similar, but I've been quite busy lately so I may not get to it any time soon.

**Comment by fescobar** [comment]
@felixp8 everything would be easy if when we dispatch the event, the API would return the RUN_ID instead of having a `204 NO CONTENT`.  I can't believe something simple and useful can not be added quickly from GH developers.

**Comment by chenghuang-mdsol** [comment]
> @felixp8 everything would be easy if when we dispatch the event, the API would return the RUN_ID instead of having a `204 NO CONTENT`. I can't believe something simple and useful can not be added quickly from GH developers.

I found a way by API. when you invoke a workflow,  pass anything unique as a parameter and show it and show it on the title. When you check the downstream workflow from uptream, search by that parameter on the tile.

**Comment by fescobar** [comment]
@chenghuang-mdsol 

Do you mean here?
```sh
POST
https://api.github.com/repos/owner/repo/actions/workflows/ci.yml/dispatches
{
    "ref": "main",
    "inputs": {
        "something": "my-random-id-xxxxx"
    }
}
```
but that parameter doesn't appear the response when I got all runs as a title

```sh
GET
https://api.github.com/repos/owner/repo/actions/runs

```
What endpoints do you mean specifically?

**Comment by chenghuang-mdsol** [comment]
> @chenghuang-mdsol
> 
> Do you mean here?
> 
> ```shell
> POST
> https://api.github.com/repos/owner/repo/actions/workflows/ci.yml/dispatches
> {
>     "ref": "main",
>     "inputs": {
>         "something": "my-random-id-xxxxx"
>     }
> }
> ```
> 
> but that parameter doesn't appear the response when I got all runs as a title
> 
> ```shell
> GET
> https://api.github.com/repos/owner/repo/actions/runs
> ```
> 
> What endpoints do you mean specifically?

In your ci.yml, do something like:
```
jobs:
  job1:
    name: Your job name - ${{ github.event.inputs.something}} 
```
Then you can search "my-random-id-xxxxx" from jobs.name, from your API response https://api.github.com/repos/owner/repo/actions/runs

I would suggest you do
https://api.github.com/repos/owner/actions/runs?branch={branch_name}&event=workflow_dispatch
Then you filter by trigger time and search key word "my-random-id-xxxxx"

**Comment by fescobar** [comment]
@chenghuang-mdsol I got it. I will try that tomorrow. Thank you!

**Comment by fescobar** [comment]
@chenghuang-mdsol I can't parameterize the name of the workflow (that would be ideal), but as you said I can parameterize the name of the job. After that, I can use this endpoint to check the name of the job inside the run workflow:
https://api.github.com/repos/owner/repo/actions/runs/RUN_ID/jobs
Thank you for your help!

**Comment by felixp8** [comment]
@chenghuang-mdsol Very neat, thanks for the tip! I'll need to try that myself
